File: Labs/Lab4/0313-Lab4.py

class TreeNode:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None



def invertTree(root):

    # need to swap every left branch with its right branch so
    # need to go top down, swapping left with the same position on the right
    # stop if it there are no children
    if (root == None): return root
    invertTree(root.left)
    invertTree(root.right)
    temp_side = root.left
    root.left = root.right
    root.right = temp_side


    return root

File: Labs/Lab4/0313-Lab4.java
package Labs.Lab4;

public class TreeNode {

    int val;
    TreeNode left;
    TreeNode right;

    public TreeNode invertTree (TreeNode root) {

        if (root == null) return null;
        TreeNode invertLeft = invertTree(root.left);
        TreeNode invertRight = invertTree(root.right);
        TreeNode temp = root.left;
        root.left = invertLeft;
        root.right = invertRight;
        return root;
    }//invert Tree
}//treenode